src/pinch-zoom.js

```diff
@@ -759,7 +760,16 @@ var definePinchZoom = function () {
          */
         disable: function() {
           this.enabled = false;
+        },
+
+        /**
+         * Unmounts the zooming container and global event listeners
+         */
+        destroy: function () {
+          window.removeEventListener('resize', this.resizeHandler);
```

Comment:
If the instance is destroyed while image loading is in flight (or possibly other async stuff), we may have callbacks triggered on the now-destroyed instance.

Not a blocker, but something to be aware of.

Reply:
I don't think there's any async handling on the `pinch-zoom-js` side that we need to care about :thinking:. As soon as the `PinchZoom` instance is created, it **_synchronously_** mounts ([`this.setupMarkdown()`](https://github.com/manuelstofer/pinchzoom/blob/master/src/pinch-zoom.js#L100)) all the required dom elements with events listeners ([`this.bindEvents()`](https://github.com/manuelstofer/pinchzoom/blob/master/src/pinch-zoom.js#L101)).

Therefore, IMHO, it's the responsibility of the library users to destroy the `PinchZoom` instance at the right time. Because only they can know _how_ the image is asynchronously loaded in different ways. E.g.: plain `img` tag with `onload` handler or some more complicated images library with a different loading handler.

Reply:
Good points, I agree!